web/src/pill_typeahead.ts

```diff
@@ -24,6 +25,18 @@ function group_matcher(query: string, item: UserGroupPillData): boolean {
     return typeahead_helper.query_matches_name(query, item);
 }
 
+// If it's only a stream search, there is no need to start the query with '#'
+function get_include_streams_function(
+    stream: boolean | undefined,
+    include_users: boolean,
+    include_user_groups?: boolean,
+): (query: string) => boolean | undefined {
+    const stream_query: (query: string) => boolean | undefined = (query: string) =>
+        stream && query.trim().startsWith("#");
+
+    return include_users || include_user_groups ? stream_query : () => stream;
+}
+
```

Comment:
This function just returns a function, and is only used at one place. It might better to just define the function directly when assigning `include_streams`

Reply:
Fixed.